test/powershell/Modules/Microsoft.PowerShell.Management/Start-Process.Tests.ps1

```diff
@@ -206,4 +206,28 @@ Describe "Start-Process" -Tags "Feature" {
             $env:TestEnvVariable = $null
         }
     }
+
+    It '-Environment adds or replaces environment variables to child process' {
+        $outputfile = Join-Path -Path $TestDrive -ChildPath output.txt
+        Start-Process pwsh -ArgumentList '-NoProfile','-Nologo','-OutputFormat xml','-Command get-childitem env:' -Wait -Environment @{ a = 1; B = 'hello'; TERM = 'dumb'; PATH = 'mine' } -RedirectStandardOutput $outputfile
+        $out = Import-Clixml $outputfile
+        ($out | Where-Object { $_.Name -eq 'a' }).Value | Should -Be 1
+        ($out | Where-Object { $_.Name -eq 'B' }).Value | Should -BeExactly 'hello'
+        ($out | Where-Object { $_.Name -eq 'TERM' }).Value | Should -BeExactly 'dumb'
+        $pathSeparator = [System.IO.Path]::PathSeparator
+        if ($IsWindows) {
+            ($out | Where-Object { $_.Name -eq 'PATH' }).Value | Should -BeLike "*${pathSeparator}mine${pathSeparator}*"
+        } else {
+            ($out | Where-Object { $_.Name -eq 'PATH' }).Value | Should -BeLike "*${pathSeparator}mine"
+        }
+    }
+
+    It '-Environment can remove an environment variable from child process' {
+        $outputfile = Join-Path -Path $TestDrive -ChildPath output.txt
+        Start-Process pwsh -ArgumentList '-NoProfile','-Nologo','-OutputFormat xml','-Command get-childitem env:' -Wait -Environment @{ a = $null; TERM = $null; TEMP = $null } -RedirectStandardOutput $outputfile
+        $out = Import-Clixml $outputfile
+        $out | Where-Object { $_.Name -eq 'a' } | Should -BeNullOrEmpty
+        $out | Where-Object { $_.Name -eq 'TERM' } | Should -BeNullOrEmpty
+        $out | Where-Object { $_.Name -eq 'TEMP' } | Should -BeNullOrEmpty
```

Comment:
Do we have to create these variables before we start the process?

Reply:
The `a` variable verifies nothing bad happens if `a` isn't already defined, `TERM` is defined on non-Windows and `TEMP` is defined on Windows.

Reply:
I suggest explicitly create `a=1` and make sure we can delete it.

Reply:
I'll update this to make it more clear what the test is intended